vfs/vfscache/cache.go

```diff
@@ -75,17 +81,21 @@ type AddVirtualFn func(remote string, size int64, isDir bool) error
 // context passed in.
 func New(ctx context.Context, fremote fs.Fs, opt *vfscommon.Options, avFn AddVirtualFn) (*Cache, error) {
 	fName := fremote.Name()
-	fRoot := filepath.FromSlash(fremote.Root())
+	fRoot := fremote.Root()
 	if runtime.GOOS == "windows" {
-		if strings.HasPrefix(fRoot, `\\?`) {
-			fRoot = fRoot[3:]
-		}
-		fRoot = strings.Replace(fRoot, ":", "", -1)
-		// Replace leading ':' if remote was created on the fly as ":backend:/path" as it is illegal in Windows
+		// Make remote name valid as dir name below cache dir.
+		// If remote was created on the fly as ":backend:/path" we must remove the leading
+		// ':', illegal character in Windows. Replacing it with something unlikely to cause
+		// collision, and something more convenient than the "fullwidth colon" encoder.OS uses.
 		if fName[0] == ':' {
 			fName = "^" + fName[1:]
 		}
+		// Make remote root valid as relative dir path below remote name dir.
+		// If it is a local path: If prefix "//?/" then remove it, and if drive
+		// letter then remove the ':' following the letter.
+		fRoot = winCachePathRegex.ReplaceAllString(fRoot, "$1$2")
```

Comment:
I don't really understand what this whole `if runtime.GOOS == "windows"` branch is about...

Cant we just do this on all OSes now?

```
fRoot := filepath.FromSlash(fremote.Root())
fRoot = toOSPath(fRoot)
```

Reply:
For a local path we need to chop off the `\\?\C:\` prefix before appending it to root path of the cache.

E.g. for command `rclone mount "C:\Temp\TEMP" X:`
The local path `C:\Temp\TEMP`, first gets translated to `//?/C:/Temp/TEMP` by rclone. Then this windows-specific block in cache makes that into `C/Temp/TEMP`, and appends that to the cache root, converting UNC, slash etc. into final result, e.g.  `\\?\C:\Rclone\data\vfs\local\C\Temp\TEMP`

So I think this is something that must be done..

Reply:
Oh, now i see: You probably meant we could just rely on the os encoder to transform `C:` into "C" followed by the fullwidth colon? That should work. Must test to see what happens, and how the `\\?\` prefix will be "encoded"..

Reply:
Yes that was what I meant - just use the path. It might look a little odd so probably worth a check!